autofs-5.1.0 - ignore multiple commas in options strings

From: Ian Kent <ikent@redhat.com>

When parsing options strings (such as opts, addopts or remopts) the parser
would object to multiple occurrances of the comma character. But this is
probably not a significant problem and the parser should just ignore them.
---
 CHANGELOG           |    1 +
 modules/amd_parse.y |    4 ++++
 modules/amd_tok.l   |    4 ++--
 3 files changed, 7 insertions(+), 2 deletions(-)

--- autofs-5.0.7.orig/CHANGELOG
+++ autofs-5.0.7/CHANGELOG
@@ -153,6 +153,7 @@
 - fix memory leak in get_defaults_entry().
 - fix out of order clearing of options buffer.
 - fix reset amd lexer scan buffer.
+- ignore multiple commas in options strings.
 
 25/07/2012 autofs-5.0.7
 =======================
--- autofs-5.0.7.orig/modules/amd_parse.y
+++ autofs-5.0.7/modules/amd_parse.y
@@ -465,6 +465,10 @@ options: OPTION
 	{
 		prepend_opt(opts, $1);
 	}
+	| OPTION COMMA
+	{
+		prepend_opt(opts, $1);
+	}
 	;
 
 %%
--- autofs-5.0.7.orig/modules/amd_tok.l
+++ autofs-5.0.7/modules/amd_tok.l
@@ -284,9 +284,9 @@ CUTSEP		(\|\||\/)
 		return SPACE;
 	}
 
-	":=" { return OPTION_ASSIGN; }
+	(:=)(,+)? { return OPTION_ASSIGN; }
 
-	"," { return COMMA; }
+	,+ { return COMMA; }
 
 	{OPTS} {
 		amd_copy_buffer();
